pkg/controller/daemon/daemoncontroller.go

```diff
@@ -703,18 +703,22 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *api.Node, ds *exte
 
 	nodeInfo := schedulercache.NewNodeInfo(pods...)
 	nodeInfo.SetNode(node)
-	fit, err := predicates.GeneralPredicates(newPod, nil, nodeInfo)
+	fit, reasons, err := predicates.GeneralPredicates(newPod, nil, nodeInfo)
 	if err != nil {
-		if re, ok := err.(*predicates.PredicateFailureError); ok {
+		message := fmt.Sprintf("GeneralPredicates failed due to %v.", err)
+		glog.Warningf("Predicate failed on Pod %s - %s", newPod.Name, message)
+	}
+	for _, r := range reasons {
+		if re, ok := r.(*predicates.PredicateFailureError); ok {
 			message := re.Error()
 			glog.V(2).Infof("Predicate failed on Pod: %s, for reason: %v", newPod.Name, message)
-		}
-		if re, ok := err.(*predicates.InsufficientResourceError); ok {
+		} else if re, ok := r.(*predicates.InsufficientResourceError); ok {
```

Comment:
Since you are already touching this code, to unify with all other places in the whole codebase, can you please change the glog messages to:

glog.V(2).Infof("... reason: %v", newPod.Name, err)

(to avoid creating those "message" strings doesn't doesn't give us anything).